Source/Core/Core/HW/DVD/AMMediaboard.cpp

```diff
@@ -0,0 +1,1841 @@
+                         data_u32[1], data_u32[2], data_u32[3]);
+        }
+
+        s_media_buffer_32[1] = 0;
+      }
+      break;
+      case AMMBCommand::Unknown_608:
+      {
+        const u32 ip = s_media_buffer_32[10];
+        u16 port = Common::swap16(s_media_buffer[6] | s_media_buffer[7] << 8);
+        u16 flag = s_media_buffer[10] | s_media_buffer[11] << 8;
+
+        NOTICE_LOG_FMT(AMMEDIABOARD_NET, "GC-AM: 0x608( {} {} {} )", ip, port, flag);
+      }
+      break;
+      case AMMBCommand::Unknown_614:
+        NOTICE_LOG_FMT(AMMEDIABOARD_NET, "GC-AM: 0x614");
+        break;
+      default:
+        ERROR_LOG_FMT(AMMEDIABOARD, "GC-AM: execute buffer UNKNOWN:{0:04x}",
+                      static_cast<u16>(ammb_command));
+        break;
+      }
+
+      memset(s_media_buffer + 0x20, 0, 0x20);
+      return 0;
+    }
+
+    PanicAlertFmtT("Unhandled Media Board Execute:{0:04x}", static_cast<u16>(ammb_command));
+    break;
+  }
+  default:
+    PanicAlertFmtT("Unhandled Media Board Command:{0:02x}", command);
+    break;
+  }
+
+  return 0;
+}
+
+u32 GetMediaType()
+{
+  switch (GetGameType())
+  {
+  default:
+  case FZeroAX:
+  case VirtuaStriker3:
+  case VirtuaStriker4:
+  case VirtuaStriker4_2006:
+  case GekitouProYakyuu:
+  case KeyOfAvalon:
+    return GDROM;
+    break;
+
+  case MarioKartGP:
+  case MarioKartGP2:
+  case FZeroAXMonster:
+    return NAND;
+    break;
+  }
+  // Never reached
+}
+
+u32 GetGameType()
+{
+  u64 game_id = 0;
+
+  if (SConfig::GetInstance().GetGameID().length() > 6)
+  {
+    game_id = 0x32323232;
+  }
+  else
+  {
+    // Convert game ID into hex
+    sscanf(SConfig::GetInstance().GetGameID().c_str(), "%s", reinterpret_cast<char*>(&game_id));
+  }
+
+  // This is checking for the real game IDs (See boot.id within the game)
+  switch ((Common::swap32((u32)game_id) >> 8) & 0xFFFF)
+  {
+  // SBGG - F-ZERO AX
+  case 0x4747:
+    return FZeroAX;
+  // SBHA - F-ZERO AX (Monster)
+  case 0x4841:
+    return FZeroAXMonster;
+  // SBKJ/SBKP - MARIOKART ARCADE GP
+  case 0x4B50:
+  case 0x4B5A:
+    return MarioKartGP;
+  // SBNJ/SBNL - MARIOKART ARCADE GP2
+  case 0x4E4A:
+  case 0x4E4C:
+    return MarioKartGP2;
+  // SBEJ/SBEY - Virtua Striker 2002
+  case 0x454A:
+  case 0x4559:
+    return VirtuaStriker3;
+  // SBLJ/SBLK/SBLL - VIRTUA STRIKER 4 Ver.2006
+  case 0x4C4A:
+  case 0x4C4B:
+  case 0x4C4C:
+    return VirtuaStriker4_2006;
+  // SBHJ/SBHN/SBHZ - VIRTUA STRIKER 4 VER.A
+  case 0x484A:
+  case 0x484E:
+  case 0x485A:
+  // SBJA/SBJJ  - VIRTUA STRIKER 4
+  case 0x4A41:
+  case 0x4A4A:
+    return VirtuaStriker4;
+  // SBFX/SBJN - Key of Avalon
+  case 0x4658:
+  case 0x4A4E:
+    return KeyOfAvalon;
+  // SBGX - Gekitou Pro Yakyuu (DIMM Upgrade 3.17)
+  case 0x4758:
+    return GekitouProYakyuu;
+  default:
+    PanicAlertFmtT("Unknown game ID:{0:08x}, using default controls.", game_id);
+  // GSBJ/G12U - VIRTUA STRIKER 3
```

Comment:
```suggestion
  [[fallthrough]];
  // GSBJ/G12U - VIRTUA STRIKER 3
```

(Assuming the fallthrough is intentional.)